### PR TITLE
Add model module with support for simulating state space models

### PR DESCRIPTION
**Pull request by shadaj**

### CODE REVIEW

File: model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package com.lynbrookrobotics.potassium.model.examples
+
+import breeze.linalg.{DenseMatrix, DenseVector}
+import breeze.plot._
+import com.lynbrookrobotics.potassium.model.StateSpace
+
+object FallingMass extends App {
+  val NumIterations = 1000
+
+  val times = new Array[Double](NumIterations)
```
Comment (PhilipAxelrod) [inline model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala]: nit, align equal signs

File: model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  val positions = new Array[Double](NumIterations)
+
+  val Mass = 1 // kg
```
Comment (PhilipAxelrod) [inline model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala]: Why is this not using squant units?

Comment (PhilipAxelrod) [reply model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala]: also, nit: first letter should be lower case for a variable name

Comment (shadaj) [reply model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala]: Good catch! I'll fix that.

File: model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  val GravityForce = Mass * 9.8 // N
```
Comment (PhilipAxelrod) [inline model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala]: Why is this not using squant units?
nit: first letter should be lower case for a variable name

File: model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      (0D, 1D),
+      (0D, 0D)
+    ),
+    DenseMatrix(
+      0,
+      -GravityForce
+    )
+  )
+
+  var state = DenseVector(0D, 0D)
```
Comment (PhilipAxelrod) [inline model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala:33]: So the first value represents time, the second position?

File: model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  for (i <- 0 until NumIterations) {
+    times(i) = i * 0.005
```
Comment (PhilipAxelrod) [inline model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala]: Where is .005 coming from?

File: model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala (hunk 3 above)
Comment (shadaj) [reply model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala]: It's a constant here, so we use camelcase

File: model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala (hunk 4 above)
Comment (shadaj) [reply model/src/test/scala/com/lynbrookrobotics/potassium/model/examples/FallingMass.scala:33]: First is position, second is velocity.

**Review by PhilipAxelrod — APPROVED** [review]
lgtm

**Pull request merged**